### ISSUE TITLE
test_msiRemoveUserFromGroup_returns_error_when_attempting_to_remove_user_from_group_which_they_are_not_a_member_of__issue_7165 fails with PREP

### ISSUE THREAD

**Issue by SwooshyCueb** [irods/irods#7455]
- [x] main
- [x] 4-3-stable

---

The test `test_all_rules.Test_AllRules.test_msiRemoveUserFromGroup_returns_error_when_attempting_to_remove_user_from_group_which_they_are_not_a_member_of__issue_7165` fails with PREP configured.
I can consistently reproduce this issue with both irods and PREP on tip of main.

Here's a couple of snippets from the test output:
```
 --- IrodsSession: icommand executed by [otherrods#tempZone] [irule -r irods_rule_engine_plugin-python-instance 
def issue_7165(_, callback, _):
    callback.msiRemoveUserFromGroup('issue_7165', 'alice', '')
 null ruleExecOut] --- 
Assert Command: irule -r irods_rule_engine_plugin-python-instance 
def issue_7165(_, callback, _):
    callback.msiRemoveUserFromGroup('issue_7165', 'alice', '')
 null ruleExecOut
Expecting STDERR: ['-819000 CAT_SUCCESS_BUT_WITH_NO_INFO']
  stdout:
    | 
  stderr:
    | remote addresses: 10.200.0.2 ERROR: rcExecMyRule error.  status = -1828000 RULE_ENGINE_ERROR
Output not found

FAILED TESTING ASSERTION
```
```
======================================================================
FAIL: test_msiRemoveUserFromGroup_returns_error_when_attempting_to_remove_user_from_group_which_they_are_not_a_member_of__issue_7165 (irods.test.test_all_rules.Test_AllRules)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/irods/scripts/irods/test/test_all_rules.py", line 1670, in test_msiRemoveUserFromGroup_returns_error_when_attempting_to_remove_user_from_group_which_they_are_not_a_member_of__issue_7165
    self.admin.assert_icommand(['irule', '-r', rep_instance, rule_map[plugin_name], 'null', 'ruleExecOut'], 'STDERR', ['-819000 CAT_SUCCESS_BUT_WITH_NO_INFO'])
  File "/var/lib/irods/scripts/irods/test/session.py", line 164, in assert_icommand
    return assert_command(*args, **kwargs)
  File "/var/lib/irods/scripts/irods/test/command.py", line 77, in assert_command
    return _assert_helper(*args, should_fail=False, **kwargs)
  File "/var/lib/irods/scripts/irods/test/command.py", line 105, in _assert_helper
    assert result
AssertionError
```

**Comment by alanking** [comment]
This test fails for PREP because `CAT_SUCCESS_BUT_WITH_NO_INFO` errors are not returned from PREP and that is the error that the test is expecting in the output.

Here is where the error number is specifically filtered out:
https://github.com/irods/irods_rule_engine_plugin_python/blob/aa6e2384ba59dd80794826e86e16beb3c32ce611/src/main.cpp#L279-L280

And here is where the error would have been returned:
https://github.com/irods/irods_rule_engine_plugin_python/blob/aa6e2384ba59dd80794826e86e16beb3c32ce611/src/main.cpp#L303-L308

So I guess we have a few options...

1. Return a different error for this case (I don't see anything fitting, so probably would need to create a new error).
2. Skip the test for PREP. 
3. Do not return/expect an error in this case (no-op) for any REP.
4. Modify PREP to return `CAT_SUCCESS_BUT_WITH_NO_INFO` errors.

**Comment by alanking** [comment]
Also, I think this is a duplicate (or at least a manifestation) of https://github.com/irods/irods/issues/7381.

**Comment by alanking** [comment]
We discussed this and decided that there are 2 steps:

1. For 4.3.2, we will go with option 1 above: Return a new error code called `USER_NOT_IN_GROUP` when attempting to remove a user from a group of which it is not a member. This seemed least offensive from a user perspective because an error is still returned even if it is different and the behavior is now consistent for the microservice across REPs.
2. Option 4 will be implemented as part of 5.0.0 and I think that #7381 should be our representative issue for that problem.
